### PR TITLE
Updating to purescript 0.7.0

### PR DESCRIPTION
**Pull request by SimonRichardson**
Tests are currently failing, need to investiagte why.

Currently :
- Have to jump through hoops because of `Data.List` vs `Data.Array`. It
   might be wiser to just move to `Data.List` and be done with it.

### CODE REVIEW

**Comment by jdegoes** [comment]
@garyb Can you review / merge / cut a new release?

@SimonRichardson Thanks!

File: bower.json [hunk 1]
```diff
@@ -17,10 +17,12 @@
     "package.json"
   ],
   "dependencies": {
-    "purescript-string-parsers": "~0.4.1",
-    "purescript-pathy": "~0.1.3",
-    "purescript-maps": "~0.3.4",
-    "purescript-integers": "~0.1.0",
-    "purescript-globals": "~0.1.6"
+    "purescript-string-parsers":  "^0.5.0",
+    "purescript-pathy":           "SimonRichardson/purescript-pathy#meltdown",
```
Comment (garyb) [inline bower.json]: Please can you update this now there's a new `pathy` release?

**Comment by SimonRichardson** [comment]
> Tests are currently failing, need to investigate why.

@garyb I need to investigate why this is happening, might get time this weekend, but unsure!

**Comment by garyb** [comment]
Ok, cool! I can't see anything obviously wrong in the updates you made. If I get around to it before you and figure it out I'll merge this anyway and then apply the fixes.

**Comment by SimonRichardson** [comment]
I've fixed one of the issues, the fact that the port was rendered as a number and not a int (easy fix), but I'm currently working out why we fail when running the parser, even though it looks correct.

**Comment by garyb** [comment]
Which case(s) is it failing on?

**Comment by SimonRichardson** [comment]
So this is the output from one of them.

```
Trying to parse telnet://192.0.2.16:80/
      printURI: telnet://192.0.2.16:80/
          show: Left (URI (Just (URIScheme "telnet")) (HierarchicalPart (Just (Authority (Nothing) [Tuple (IPv4Address "192.0.2.16") (Just (80))])) (Just (Right (rootDir)))) (Nothing) (Nothing))
```

**Comment by garyb** [comment]
Odd, that should be passing I think... I'll check it out and have a look too.

**Comment by garyb** [comment]
Ah, that one did pass, some of the others are failing though.

**Comment by garyb** [comment]
Oh, actually, I think these tests always failed: they're unimplemented features (email addresses, relative URLs, news/tel/urn urls that I haven't figured out which rule is being broken).

Can you group the failing tests together and `log` "Failing tests" or something before them please? That'll do for now, unless you really feel like fixing the parser for the less usual cases. :smile:

**Comment by SimonRichardson** [comment]
@garyb I think it's best to release this as is, and I'll have a play at updating the parser later on.

**Comment by garyb** [comment]
Great! Thanks for working on this.

**Comment by SimonRichardson** [comment]
sure nps.

**Comment by garyb** [comment]
It appears this wasn't released at all before, so I've released it as `v0.1.0-rc.1` for now. I'll try to fix up the parser issues before making a "proper" release.

**Pull request merged**